src/wallet/wallettool.cpp

```diff
@@ -0,0 +1,145 @@
+        if (load_wallet_ret == DBErrors::CORRUPT) {
+            fprintf(stderr, "Error loading %s: Wallet corrupted", name.c_str());
+            return nullptr;
+        } else if (load_wallet_ret == DBErrors::NONCRITICAL_ERROR) {
+            fprintf(stderr, "Error reading %s! All keys read correctly, but transaction data"
+                            " or address book entries might be missing or incorrect.",
+                name.c_str());
+        } else if (load_wallet_ret == DBErrors::TOO_NEW) {
+            fprintf(stderr, "Error loading %s: Wallet requires newer version of %s",
+                name.c_str(), PACKAGE_NAME);
+            return nullptr;
+        } else if (load_wallet_ret == DBErrors::NEED_REWRITE) {
+            fprintf(stderr, "Wallet needed to be rewritten: restart %s to complete", PACKAGE_NAME);
+            return nullptr;
+        } else {
+            fprintf(stderr, "Error loading %s", name.c_str());
+            return nullptr;
+        }
+    }
+
+    return wallet_instance;
+}
+
+static void WalletShowInfo(CWallet* wallet_instance)
+{
+    // lock required because of some AssertLockHeld()
+    LOCK(wallet_instance->cs_wallet);
+
+    fprintf(stdout, "Wallet info\n===========\n");
+    fprintf(stdout, "Encrypted: %s\n", wallet_instance->IsCrypted() ? "yes" : "no");
+    fprintf(stdout, "HD (hd seed available): %s\n", wallet_instance->GetHDChain().seed_id.IsNull() ? "no" : "yes");
+    fprintf(stdout, "Keypool Size: %u\n", wallet_instance->GetKeyPoolSize());
+    fprintf(stdout, "Transactions: %zu\n", wallet_instance->mapWallet.size());
+    fprintf(stdout, "Address Book: %zu\n", wallet_instance->mapAddressBook.size());
+}
+
+bool executeWalletToolFunc(const std::string& method, const std::string& name)
+{
+    fs::path path = fs::absolute(name, GetWalletDir());
+
+    if (method == "create") {
+        std::shared_ptr<CWallet> wallet_instance = CreateWallet(name, path);
+        if (wallet_instance) {
+            WalletShowInfo(wallet_instance.get());
+            wallet_instance->Flush();
+        }
+    } else if (method == "info") {
+        std::shared_ptr<CWallet> wallet_instance = LoadWallet(name, path);
+        if (!wallet_instance) return false;
+        WalletShowInfo(wallet_instance.get());
+        wallet_instance->Flush();
+    } else if (method == "salvage") {
+        // Recover readable keypairs:
+        std::string error;
+        if (!WalletBatch::VerifyEnvironment(path, error)) {
+            fprintf(stderr, "WalletBatch::VerifyEnvironment Error: %s\n", error.c_str());
+            return false;
+        }
+
+        CWallet dummyWallet("dummy", WalletDatabase::CreateDummy());
+        std::string backup_filename;
+        if (!WalletBatch::Recover(path, (void*)&dummyWallet, WalletBatch::RecoverKeysOnlyFilter, backup_filename)) {
+            fprintf(stderr, "Salvage failed\n");
+            return false;
+        }
+        //TODO, set wallets best block to genesis to enforce a rescan
```

Comment:
```
2018-09-22 20:39:03 cpplint(pr=14284): src/wallet/wallettool.cpp:125:  Should have a space between // and comment  [whitespace/comments
```

Reply:
This is not my commit, please comment on #13926.